src/marketplace/ListingsView.js

```diff
@@ -69,34 +74,26 @@ class ListingsView extends React.Component<Props, State> {
 			const typeOfItem = getTankComponent(verifyComponent(jsonObjectOfSells[sale].itemId));
 			if(typeOfItem === this.props.sellerType)
 			{
-				//This is a temp obj used to create an object of what a sell item would need to have. It is pushed into an array of all of the items for sell that fall in the current seller type
-				let obj = {};
-				obj['name'] = jsonObjectOfSells[sale].itemId;
-				obj['price'] = jsonObjectOfSells[sale].salePrice;
-				obj['amount'] = jsonObjectOfSells[sale].amount;
-				obj['sellerId'] = jsonObjectOfSells[sale].sellerId;
-				obj['saleId'] = jsonObjectOfSells[sale]._id;
-				itemsForSale.push(obj);
+				const sellingObject = new saleObject(jsonObjectOfSells[sale].itemId,jsonObjectOfSells[sale].salePrice,jsonObjectOfSells[sale].amount,jsonObjectOfSells[sale].sellerId,jsonObjectOfSells[sale]._id);
+				itemsForSaleArray.push(sellingObject);
 			}
 		}
-		//Need this to deal with the asynch nature of api calling......fun times
-		//Makes sure that we create the cards after we get the sells
-		this.forceUpdate();     
+		this.setState({itemsForSale:itemsForSaleArray});  
 	}
 	
 	//This creates a card for every sale
 	createCards = () => {
-		let cards = []
+		const cards = []
 		// Outer loop to create parent
-		for (let i = 0; i < itemsForSale.length; i++) {
+		for (let i = 0; i < this.state.itemsForSale.length; i++) {
 			//Create the parent and add the children
 			cards.push(
 				<div className="card mb-2" key={i}>
 					<div className="card-body">
-						<h5 className="card-title">Item to buy: {itemsForSale[i].name}</h5>
-						<h5 className="card-title">Price: ${itemsForSale[i].price}</h5>
-						<h5 className="card-title">Quantity: {itemsForSale[i].amount}</h5>
-						<button className="btn btn-success mt-2" onClick={this.buyItem.bind(this,itemsForSale[i].sellerId, itemsForSale[i].saleId)}>Buy</button>
+						<h5 className="card-title">Item to buy: {this.state.itemsForSale[i].name}</h5>
+						<h5 className="card-title">Price: ${this.state.itemsForSale[i].price}</h5>
+						<h5 className="card-title">Quantity: {this.state.itemsForSale[i].amount}</h5>
+						<button className="btn btn-success mt-2" onClick={this.buyItem.bind(this,this.state.itemsForSale[i].sellerId, this.state.itemsForSale[i].saleId)}>Buy</button>
```

Comment:
You can make this line cleaner and easier to read with ES6:
`onClick={() => this.buyItem(this.state.itemsForSale[i].sellerId, this.state.itemsForSale[i].saleId)}`